### PR TITLE
[TorchToTosa] max_pool3d legalization

### PR DESCRIPTION
**Pull request by ahagro01**
Add Torch-to-TOSA legalization for torch.aten.max_pool3d.

The lowering decomposes 3D max pooling into two tosa.max_pool2d operations:
1. Pool the height and width dimensions while folding batch and depth together.
2. Pool the depth dimension as a D×1 window while folding batch and the output spatial dimensions together.
3. Reshape and transpose the result back to NCDHW or CDHW layout.
The implementation shares the existing pooling input preparation logic to handle padding, ceil mode, and trailing input slicing.

Current limitations
- Input shape must be static.
- Input must be rank 4 or rank 5.
- Kernel, stride, padding, dilation, and ceil mode must be constant.
- Only unit dilation is supported.

Testing
- Added Torch-to-TOSA conversion tests covering:
  - Basic MaxPool3d lowering.
  - Floor mode with trailing input slicing.
  - Unbatched input with padding and ceil mode.
- Enabled the supported static MaxPool3d E2E tests for the FX-importer TOSA configuration.

### CODE REVIEW

File: lib/Conversion/TorchToTosa/TorchToTosa.cpp [hunk 1]
```diff
@@ -8945,6 +8973,169 @@ class ConvertAtenMaxPool1dOp
...
+    auto kernelAttr = rewriter.getDenseI64ArrayAttr(kernel);
+    auto strideAttr = rewriter.getDenseI64ArrayAttr(stride);
+    auto padAttr = rewriter.getDenseI64ArrayAttr(
+        {padding[0], padding[0], padding[1], padding[1]});
+    auto prepared = preparePoolingInput(rewriter, loc, input, kernelAttr,
+                                        strideAttr, padAttr, {1, 1}, ceilMode);
+
+    auto outputTy =
+        RankedTensorType::get({inputShape[0], prepared.outputShape[0],
+                               prepared.outputShape[1], inputShape[3]},
```
Comment (Lallapallooza) [inline lib/Conversion/TorchToTosa/TorchToTosa.cpp:9161]: The example below returns an output depth of 4 in PyTorch, but this lowering computes 3 and fails on the final cast. `preparePoolingInput` reduces the trailing padding from 2 to 1, dropping the last window even though it still overlaps the input. Can we fix the ceil-mode padding adjustment and add a regression test?

```python
import torch
from torch_mlir import fx

model = torch.nn.MaxPool3d(
    kernel_size=(5, 1, 1),
    stride=(2, 1, 1),
    padding=(2, 0, 0),
    ceil_mode=True,
)
x = torch.arange(6, dtype=torch.float32).reshape(1, 1, 6, 1, 1)

expected = model(x)
print("PyTorch shape:", tuple(expected.shape), flush=True)
print("PyTorch values:", expected.flatten().tolist(), flush=True)

exported = torch.export.export(model, (x,), strict=True)
tosa_module = fx.export_and_import(exported, output_type="tosa")
print(tosa_module)
```

**Review by Lallapallooza — APPROVED** [review]
LGTM after a small fix

File: lib/Conversion/TorchToTosa/TorchToTosa.cpp (hunk 1 above)
Comment (ahagro01) [reply lib/Conversion/TorchToTosa/TorchToTosa.cpp:9161]: Fixed the ceil-mode padding adjustment to preserve the final partial window. To match PyTorch semantics, a window is excluded when its entire kernel lies within the right padding. Added regression tests covering both overlapping and fully padded final windows.
